src/main/java/com/FlightPub/RequestObjects/BasicSearch.java

```diff
@@ -166,8 +167,8 @@ public List<StopOver> basicStopOverSearch(int numberOfStops) {
             List<StopOver> output = new ArrayList<>();  // Stores the stopovers of each stopover iteration
             for (StopOver flight : flights) {
                 for (Flight newFlight : allFlights) {
-                    if (flight.getFlightAtIndex(count).getDestinationID().equals(newFlight.getOriginID())) {
-                        if (flight.locationVisited(newFlight.getDestinationID()) == false) {
+                    if (flight.getFlightAtIndex(count).getDestinationCode().equals(newFlight.getDepartureCode())) {
```

Comment:
This should be newFlight.getOriginID() since we are checking they are meeting at a common airport